components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -85,6 +92,29 @@ net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
     )");
 }
 
+// https://tools.ietf.org/html/rfc1035#section-3.1
+absl::optional<std::vector<uint8_t>> DnsEncode(const std::string& dotted_name) {
```

Comment:
is there not something in Chromium we can use to do this instead of writing it ourselves? cc @bridiver

Reply:
There is similar thing in chrome https://source.chromium.org/chromium/chromium/src/+/main:net/dns/dns_util.cc;l=47?q=DNSDomainFromDot&sq=&ss=chromium
But per ENS spec https://docs.ens.domains/ens-improvement-proposals/ensip-10-wildcard-resolution#specification there is no total length limitation, so I needed a custom converter.

Will try to check if >256 char length domains actually exist in ENS, if not maybe it makes sense to use chrome's implementation.

Still needs a better comment here.